sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -800,6 +804,61 @@ class CollationStringExpressionsSuite
     assert(collationMismatch.getErrorClass === "COLLATION_MISMATCH.EXPLICIT")
   }
 
+  test("Support mode for string expression with collation") {
+    val query = "SELECT mode(collate('abc', 'utf8_binary'))"
+    checkAnswer(sql(query), Row("abc"))
+    assert(sql(query).schema.fields.head.dataType.sameType(StringType("utf8_binary")))
+  }
```

Comment:
you can remove this test, as it doesn't seem to do anything - i.e. we already know that mode supports regular strings (UTF8_BINARY)